### ISSUE TITLE
Extremely slow response for large project

### ISSUE THREAD

**Issue by StarSkiandHike** [texstudio-org/texstudio#812]
### Environment
- TeXstudio: 2.12.16
- Qt: 5.12.1
- OS: Windows 10 Pro
- TeX distribution: MiKTeX 2.9

### Actual behavior

I am writing a scientific text that is approaching 1000 pages.  Starting a couple months ago performance has almost died.  Just starting TXS can take 10 minutes.  At times if I change a setting (even something as simple as setting show all line numbers), type a LaTeX command, or load a new file, many minutes, and sometimes hours go by before TXS finally finishes the request.  When I check CPU usage, it is clearly processing, typically with 25% of the 4 CPUs, and memory working set values of about 500,000 kB.  There are no hard faults and very little to no disk writing. 

I have completely uninstalled both TXS and MiKTeX multiple times, thoroughly cleaned the registry, and deleted temporary directories.  When reinstalling I make sure that I am using the latest stable releases.

I seemed to have made some improvement with typing latex commands by creating my own .cwl packages for  my own .sty files, and forced TXS to load all .cwl files for all packages that I am using, but it doesn't always help, and certainly doesn't impact simple settings changes.

**Comment by foreachthing** [comment]
My two cents:

Try to split up your document, like chapter001.tex, chapter002.tex ... and `\include` those files in your main document. Read more here: https://tex.stackexchange.com/questions/29577/splitting-a-large-document-into-several-files

One advantage: You can comment entire chapters/sections and compile only the one(s) you're currently working on. Also you use the `draft` option `\documentclass[draft]{ yourDOCCLASS }` if you have lots of pictures.

Maybe even "forget" about layout and lots of packages. Type your text first. After your done, do your layout.
Write your layout on a small section of your paper and then use it on the whole.

**Comment by dbitouze** [comment]
> One advantage: You can comment entire chapters/sections and compile only the one(s) you're currently working on.

Much better: make use of the `\includeonly` command, such as in the following example:
```latex
\includeonly{introduction,conclusion}
\documentclass{book}
% Packages
\begin{document}
\frontmatter
\tableofcontents
\include{foreword}
\mainmatter
\include{introduction}
\include{chapter1}
\include{chapter2}
\include{chapter3}
\include{conclusion}
\appendix
\include{aappendices}
\backmatter
\include{afterword}
\end{document}
```

**Comment by foreachthing** [comment]
@dbitouze, OK! I didn't know about that command! That's handy! 👍

**Comment by dbitouze** [comment]
> But, I believe, that txs will still be slow ... since it's not commented it _might_ still get checked for references and ... stuff.

Indeed...

**Comment by StarSkiandHike** [comment]
Thanks for the suggestions, but I have already been using \include and \input commands for chapters and sections, respectively.  Actually, it isn't the MikTeX compiles that are the problem, it seems to be TXS's behind-the-scenes bookkeeping of the files, images, etc. along with trying to keep track of a very large  number of user-defined commands for astronomical and physical constants, units, environments, etc.  I have noticed that some pieces of MikTeX do get called even when compiling isn't happening, perhaps as part of the behind-the-scenes processes.

**Comment by sunderme** [comment]
disabling „automatically load included files“ should improve start up. Some of the background functionality can be turned off.
But all these steps reduce txs capabilities.

You could send me the source, so that I can take look whether something odd slows down things, but that is basically it.

**Comment by MeanSquaredError** [comment]
You can try disabling all the TXS options that are related to the editing and see if it will speed up work. I am not sure if it will make a difference but you can try.

These options include:

Editor / Folding
Editor / Analyze LaTeX
Editor / Analyze characters
Editor / Inline checking (I suspect this slows down a bit)
Adv. Editor / Special options / Auto complete parentheses
Adv. Editor / Special options / Overwrite opening bracket
Adv. Editor / Special options / Overwrite closing bracket
Adv. Editor / Special options / Show image tooltip
Adv. Editor / Special options / Show help ...
Adv. Editor / Structure panel (all of the options there)
Completion options (disable all of them)
Language Checking (all of the options there)
Preview / Show preview as tooltip ...

Basically anything that adds more eye-candy and make TeXstudio "smarter" slows it down a bit.

**Comment by foreachthing** [comment]
I noticed that syntax highlighting takes a lot of time ... is there an easy way to (temporarily) disable it?

**Comment by MeanSquaredError** [comment]
@foreachthing 
Unchecking the button Editor / Inline Checking turns off most of the syntax highlighting. I would guess that this would speed up the UI a bit although I did not do any profiling to measure the exact speed-up.
